notebooks/circuits_with_bloqade.py

```diff
@@ -0,0 +1,763 @@
+            op_zz(theta = dt*J, qb1 = qubits[i], qb2 = qubits[i + 1])
+        for i in range(0, len(qubits)):
+            squin.qubit.apply(operator=xpow, qubits=[qubits[i]])
+    return qubits
+
+
+# %% [markdown]
+# Of course, both Cirq and the (converted) Bloqade kernel have the same execution and same output state.
+
+# %%
+cirq_trotter = trotter_circuit(
+    N=12,
+    steps=10,
+    dt=0.01,
+    J=1,
+    h=1)
+
+cirq_statevector = cirq.Simulator().simulate(cirq_trotter).state_vector()
+
+# Or converting to a bloqade kernel and simulating with PyQrack
+cirq_trotter_kernel = squin.cirq.load_circuit(
+    cirq_trotter,
+    kernel_name="cirq_trotter",
+    register_as_argument=False,
+    return_register=True)
+cirq_trotter_qubits = StackMemorySimulator(min_qubits=12).task(cirq_trotter_kernel).run()
+cirq_bloqade_state = StackMemorySimulator.quantum_state(cirq_trotter_qubits)
+# The state is, of course, the same. A little bit of work is needed to extract out the (single) state vector from the RDM.
+print("Overlap:", np.abs(np.dot(np.conj(cirq_statevector), cirq_bloqade_state.eigenvectors[:,0]))**2)
+
+
+# %% [markdown]
+# Similarly, the execution and output state of the cirq state and kernel written fully in bloqade have the same state. Note that for the `bloqade_trotter` kernel, the arguments are not declared until the simulator is run.
+
+# %%
+# Option C: Bloqade only
+cirq_trotter_qubits = StackMemorySimulator(min_qubits=12).run(bloqade_trotter, args=(12,10,0.01,1,1))
+cirq_bloqade_state = StackMemorySimulator.quantum_state(cirq_trotter_qubits)
+print("Overlap:", np.abs(np.dot(np.conj(cirq_statevector), cirq_bloqade_state.eigenvectors[:,0]))**2)
+
+# %% [markdown]
+# # Mid-circuit feed forward
+# 
+# The powerful thing about bloqade kernels is the ability to natively represent mid-circuit feed-forward using control flow. While the possibilities are endless, including measurement based quantum computing and error correction, we show two examples here.
+# 
+# The first is T state teleportation, which teleports a T gate that was applied to an ancilla (a "T state") onto the target state using only Clifford gates and feedforward. Due to the property of being Clifford, the gadget itself is fault tolerant and thus plays an important role in many error corrected algorithms.
+
+# %%
+try:
+    del t_teleport
+except:
+    pass
+@squin.kernel
+def t_teleport(target:squin.qubit.Qubit) -> squin.qubit.Qubit:
+    ancilla = squin.qubit.new(1)[0]
+    squin.gate.h(ancilla)
+    squin.gate.t(ancilla)
+    squin.gate.cx(control=target, target=ancilla)
+    bit = squin.qubit.measure(target)
+    if bit:
+        squin.gate.s(ancilla)
+    return ancilla # The state of the target qubit is also teleported to the ancilla
+
+
+# And now lets wrap it into a larger context to run. In this case,
+# apply to a |+> state and see that we get a T|+> state out.
+# https://github.com/QuEraComputing/bloqade-circuit/issues/431
```

Comment:
Why are you referencing this issue here?

Reply:
Generally, I don't think we should reference issues in tutorials. Also, it wasn't very clear to me what the context here is when I read through it, so I'm not sure it will be clear to users either.